src/components/Comment/index.spec.tsx

```diff
@@ -469,4 +468,93 @@ describe(__filename, () => {
       );
     });
   });
+
+  describe('discarding a new comment', () => {
+    it('renders a discard button to cancel entering a comment', () => {
+      const versionId = 33;
+      const fileName = 'file.name.json';
+      const line = 100;
+      const store = configureStore();
+      const payload = {
+        fileName,
+        line,
+        versionId,
+      };
+
+      store.dispatch(actions.beginComment(payload));
+      const dispatchSpy = spyOn(store, 'dispatch');
+      const root = render({
+        fileName,
+        line,
+        readOnly: false,
+        store,
+        versionId,
+      });
+      const discardButton = root.find(`.${styles.discardButton}`);
+      discardButton.simulate('click');
+
+      expect(discardButton).toHaveText('Discard');
```

Comment:
The placement of this assertion is a bit misleading. The test makes it look like you are clicking the discard button (via `discardButton.simulate('click')`) and *then* checking the text of the button, which at this point would actually be "Confirm discard" (because the button was clicked). 

If the goal is to check the text of the button as it was before it was clicked, I suggest you move this assertion above the call to `discardButton.simulate('click')`.

Another option would be to have 2 different tests, one to check that `actions.considerDiscardComment` is in fact called after clicking the button, and another to check the text of the Discard button, which wouldn't require a call to `discardButton.simulate('click')` at all.  I'm actually fine with both things being tested in a single test, but @kumar303 might feel differently.

Reply:
> The placement of this assertion is a bit misleading

Thanks for catching that. It is misleading...

Reply:
> Another option would be to have 2 different tests, one to check that `actions.considerDiscardComment` is in fact called after clicking the button, and another to check the text of the Discard button...

I think it would be fine to make those two assertions in the same test.